earthaccess/typing_.py

```diff
@@ -0,0 +1,49 @@
+"""
+Convenience module for importing types from the typing module, abstracting away
+the differences between Python versions.
+"""
+
+import sys
+from typing import Any, Callable, Optional, SupportsFloat, Type, Union, cast
+
+if sys.version_info < (3, 9):
+    from typing import Dict, List, Mapping, Sequence, Tuple
+else:
+    from builtins import dict as Dict, list as List, tuple as Tuple
+    from collections.abc import Mapping, Sequence
```

Comment:
Yes, I pulled all imports from `typing`, `typing_extensions`, and `collections.abc` into this module so that throughout the earthaccess code, we simply import from `.typing_` (or `earthaccess.typing_`) to avoid having to deal with all of the changes that have been made across Python versions.

There are various "collection" types prior to 3.9 that were exposed from the `typing` module, but as of 3.9, the recommendation is to import them from `collections.abc`. There's also the transition from using `List`, `Dict`, and `Tuple` to `list`, `dict`, and `tuple`, respectively, along with various other changes that are annoying to keep up with.

Therefore, this `typing_` module centralizes all such logic in one place, so throughout the rest of the code base, we can ignore it all and import all types from `typing_`.